src/pages/home-page-slides/slides-order/slides-order.js

```diff
@@ -0,0 +1,186 @@
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import Card from '@material-ui/core/Card';
+import Paper from '@material-ui/core/Paper';
+import Avatar from '@material-ui/core/Avatar';
+import ImageIcon from '@material-ui/icons/Image';
+import {Typography } from '@material-ui/core';
+import { useDispatch} from 'react-redux';
+import { useStyles } from './slides-order.styles';
+import { config } from '../../../configs';
+import { SaveButton } from '../../../components/buttons';
+import { slidesTranslations } from '../../../translations/home-page-slides.translations';
+import {
+  setSlides,
+  updateSlidesOrder
+} from '../../../redux/home-page-slides/home-page-slides.actions';
+
+const SlidesOrder = (props) => {
+  const styles = useStyles();
+  const dispatch = useDispatch();
+  const { drugAndDropList } = props;
+  const [list, setList] = useState(drugAndDropList);
+  const [dragging, setDragging] = useState(false);
+  const [draggable, setDraggable] = useState(false);
+  const { IMG_URL } = config;
+  const {discoverMoreTitle,
+    slideOrderTitle} = config.titles.homePageSliderTitle;
+  const {OPEN_SLIDE_EDIT,SAVE_SLIDE_ORDER} = config.buttonTitles
+  const dragItem = useRef();
+  const dragItemNode = useRef();
+
+  const handlerDragStart = (e, item) => {
+    dragItemNode.current = e.target;
+    dragItemNode.current.addEventListener('dragend', handleDragEnd);
+    dragItem.current = item;
+    setTimeout(() => {
+      setDragging(true);
+    }, 100);
+  };
+
+  const handleDragEnter = (e, targetItem) => {
+    if (dragItemNode.current !== e.target) {
+      setList(oldList => {
+        const newList = JSON.parse(JSON.stringify(oldList));
+        newList[targetItem.grpI].items
+          .splice(targetItem.itemI, 0, newList[dragItem.current.grpI]
+            .items.splice(dragItem.current.itemI, 1)[0]);
+        dragItem.current = targetItem;
+        return newList;
+      });
+    }
+  };
+
+  const handleDragEnd = () => {
+    setDragging(false);
+    dragItem.current = null;
+    dragItemNode.current.removeEventListener('dragend', handleDragEnd);
+    dragItemNode.current = null;
+  };
+
+  const getStyles = (item) => {
+    if (dragItem.current.grpI === item.grpI && dragItem.current.itemI === item.itemI) {
+      return `${styles.dndItem} ${styles.current}`;
+    }
+    return styles.dndItem;
+  };
+  const changeHandler = () => setDraggable(true);
+  const saveHandler = ()=>{
+    const availableArray = []
+    const nonAvailableArray = []
+    list.forEach(el=>{
+
+      if(el.title==='available'){
+        el.items.map((availableSlide,index) => {
+          const {order,show,...rest} = availableSlide
+          availableArray.push({id:availableSlide._id,slide:{order:+index+1, show:true, ...rest}})
+          return {order,show}
+        });
+      }
+      if(el.title==='nonAvailable'){
+        el.items.map((nonAvailableSlide) => {
```

Comment:
You could replace `map` to `forEach` because returning value of this method are ignored

Reply:
Fixed